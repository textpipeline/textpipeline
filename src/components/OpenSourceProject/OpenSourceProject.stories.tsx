import { storiesOf } from '@storybook/react';
import React from 'react';
import { testOss } from 'testing/fixtures';

import OpenSourceProject from './OpenSourceProject';

storiesOf('OpenSourceProject', module).add('default', () => (
  <OpenSourceProject
    name={testOss.name}
    description={testOss.description}
    slug={testOss.slug}
    projectHref={testOss.projectHref}
    repositoryHref={testOss.repositoryHref}
    license={testOss.license}
    licenseText={testOss.licenseText}
    transformsPath="/transforms"
  />
));
