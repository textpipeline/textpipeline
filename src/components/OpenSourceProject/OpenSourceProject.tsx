import Box from '@material-ui/core/Box';
import Divider from '@material-ui/core/Divider';
import List from '@material-ui/core/List';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import LinkIcon from '@material-ui/icons/Link';
import React from 'react';

import { License } from '../../licenses';
import { byProjectSlug } from '../../transforms';
import Heading from '../Heading';
import ListItemLink from '../ListItemLink';
import TransformList from '../TransformList';

export interface OpenSourceProjectProps {
  readonly name: string;
  readonly description: string;
  readonly slug: string;
  readonly projectHref: string;
  readonly repositoryHref: string;
  readonly license: License;
  readonly licenseText: string;
  readonly transformsPath: string;
}

const OpenSourceProject: React.FC<OpenSourceProjectProps> = ({
  name,
  slug,
  description,
  projectHref,
  repositoryHref,
  license,
  licenseText,
  transformsPath,
}) => {
  const transforms = byProjectSlug[slug] || [];
  return (
    <>
      <Box mb={2}>
        <Heading level={2}>{name}</Heading>
        <Typography variant="subtitle1">{description}</Typography>
      </Box>
      <Box display="flex">
        <Paper>
          <Box px={2}>
            <Box pb={3}>
              <List>
                <ListItemLink icon={LinkIcon} text="Project" subtitle={projectHref} href={projectHref} isExternal />
                <Divider />
                <ListItemLink
                  icon={LinkIcon}
                  text="Repository"
                  subtitle={repositoryHref}
                  href={repositoryHref}
                  isExternal
                />
                <Divider />
                <ListItemLink icon={LinkIcon} text="License" subtitle={license.name} href={license.href} isExternal />
                <Divider />
              </List>
            </Box>
            <Heading level={4}>Transforms using this project:</Heading>
            <TransformList transforms={transforms} transformsPath={transformsPath} />
          </Box>
        </Paper>
        <Box px={2}>
          <pre>{licenseText}</pre>
        </Box>
      </Box>
    </>
  );
};

export default OpenSourceProject;
