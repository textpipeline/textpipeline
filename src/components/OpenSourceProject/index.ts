export * from './OpenSourceProject';
export { default } from './OpenSourceProject';
