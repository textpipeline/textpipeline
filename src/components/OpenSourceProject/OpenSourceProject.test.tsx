import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import { testOss, testTransform as mockTransform } from '../../fixtures';
import OpenSourceProject from './OpenSourceProject';

jest.mock('../../transforms', () => ({
  byProjectSlug: {
    'test-open-source-project': [mockTransform],
  },
}));

describe('OpenSourceProject', () => {
  let rendered: ShallowWrapper;

  describe('when there are transforms for this project', () => {
    beforeEach(() => {
      rendered = shallow(
        <OpenSourceProject
          name={testOss.name}
          description={testOss.description}
          slug={testOss.slug}
          projectHref={testOss.projectHref}
          repositoryHref={testOss.repositoryHref}
          license={testOss.license}
          licenseText={testOss.licenseText}
          transformsPath="/transforms"
        />
      );
    });

    it('renders correctly', () => {
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('when there are no transforms for this project', () => {
    beforeEach(() => {
      rendered = shallow(
        <OpenSourceProject
          name={testOss.name}
          description={testOss.description}
          slug="without-transforms"
          projectHref={testOss.projectHref}
          repositoryHref={testOss.repositoryHref}
          license={testOss.license}
          licenseText={testOss.licenseText}
          transformsPath="/transforms"
        />
      );
    });

    it('renders correctly', () => {
      expect(rendered).toMatchSnapshot();
    });
  });
});
