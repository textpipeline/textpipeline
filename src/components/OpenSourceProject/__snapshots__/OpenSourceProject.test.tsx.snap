// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenSourceProject when there are no transforms for this project renders correctly 1`] = `
<Fragment>
  <Styled(MuiBox)
    mb={2}
  >
    <Heading
      level={2}
    >
      Test Open Source Project
    </Heading>
    <WithStyles(ForwardRef(Typography))
      variant="subtitle1"
    >
      Test Open Source Project Description
    </WithStyles(ForwardRef(Typography))>
  </Styled(MuiBox)>
  <Styled(MuiBox)
    display="flex"
  >
    <WithStyles(ForwardRef(Paper))>
      <Styled(MuiBox)
        px={2}
      >
        <Styled(MuiBox)
          pb={3}
        >
          <WithStyles(ForwardRef(List))>
            <ListItemLink
              href="https://foobarbazbuz.github.io"
              icon={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "displayName": "LinkIcon",
                  "muiName": "SvgIcon",
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                }
              }
              isExternal={true}
              subtitle="https://foobarbazbuz.github.io"
              text="Project"
            />
            <WithStyles(ForwardRef(Divider)) />
            <ListItemLink
              href="https://github.com/foobarbazbuz/foobarbazbuz/"
              icon={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "displayName": "LinkIcon",
                  "muiName": "SvgIcon",
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                }
              }
              isExternal={true}
              subtitle="https://github.com/foobarbazbuz/foobarbazbuz/"
              text="Repository"
            />
            <WithStyles(ForwardRef(Divider)) />
            <ListItemLink
              href="https://choosealicense.com/licenses/mit"
              icon={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "displayName": "LinkIcon",
                  "muiName": "SvgIcon",
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                }
              }
              isExternal={true}
              subtitle="MIT"
              text="License"
            />
            <WithStyles(ForwardRef(Divider)) />
          </WithStyles(ForwardRef(List))>
        </Styled(MuiBox)>
        <Heading
          level={4}
        >
          Transforms using this project:
        </Heading>
        <TransformList
          transforms={Array []}
          transformsPath="/transforms"
        />
      </Styled(MuiBox)>
    </WithStyles(ForwardRef(Paper))>
    <Styled(MuiBox)
      px={2}
    >
      <pre>
        
MIT License

Copyright (c) 2020 John Doe

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
  
      </pre>
    </Styled(MuiBox)>
  </Styled(MuiBox)>
</Fragment>
`;

exports[`OpenSourceProject when there are transforms for this project renders correctly 1`] = `
<Fragment>
  <Styled(MuiBox)
    mb={2}
  >
    <Heading
      level={2}
    >
      Test Open Source Project
    </Heading>
    <WithStyles(ForwardRef(Typography))
      variant="subtitle1"
    >
      Test Open Source Project Description
    </WithStyles(ForwardRef(Typography))>
  </Styled(MuiBox)>
  <Styled(MuiBox)
    display="flex"
  >
    <WithStyles(ForwardRef(Paper))>
      <Styled(MuiBox)
        px={2}
      >
        <Styled(MuiBox)
          pb={3}
        >
          <WithStyles(ForwardRef(List))>
            <ListItemLink
              href="https://foobarbazbuz.github.io"
              icon={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "displayName": "LinkIcon",
                  "muiName": "SvgIcon",
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                }
              }
              isExternal={true}
              subtitle="https://foobarbazbuz.github.io"
              text="Project"
            />
            <WithStyles(ForwardRef(Divider)) />
            <ListItemLink
              href="https://github.com/foobarbazbuz/foobarbazbuz/"
              icon={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "displayName": "LinkIcon",
                  "muiName": "SvgIcon",
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                }
              }
              isExternal={true}
              subtitle="https://github.com/foobarbazbuz/foobarbazbuz/"
              text="Repository"
            />
            <WithStyles(ForwardRef(Divider)) />
            <ListItemLink
              href="https://choosealicense.com/licenses/mit"
              icon={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "displayName": "LinkIcon",
                  "muiName": "SvgIcon",
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                }
              }
              isExternal={true}
              subtitle="MIT"
              text="License"
            />
            <WithStyles(ForwardRef(Divider)) />
          </WithStyles(ForwardRef(List))>
        </Styled(MuiBox)>
        <Heading
          level={4}
        >
          Transforms using this project:
        </Heading>
        <TransformList
          transforms={
            Array [
              Object {
                "defaultOutput": "0",
                "execute": [Function],
                "inputType": "Foo",
                "name": "Foo to Bar",
                "outputType": "Bar",
                "project": Object {
                  "description": "Test Open Source Project Description",
                  "license": Object {
                    "href": "https://choosealicense.com/licenses/mit",
                    "name": "MIT",
                    "slug": "mit",
                  },
                  "licenseText": "
MIT License

Copyright (c) 2020 John Doe

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the \\"Software\\"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED \\"AS IS\\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
  ",
                  "name": "Test Open Source Project",
                  "projectHref": "https://foobarbazbuz.github.io",
                  "repositoryHref": "https://github.com/foobarbazbuz/foobarbazbuz/",
                  "slug": "test-open-source-project",
                },
                "slug": "foo-to-bar",
              },
            ]
          }
          transformsPath="/transforms"
        />
      </Styled(MuiBox)>
    </WithStyles(ForwardRef(Paper))>
    <Styled(MuiBox)
      px={2}
    >
      <pre>
        
MIT License

Copyright (c) 2020 John Doe

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
  
      </pre>
    </Styled(MuiBox)>
  </Styled(MuiBox)>
</Fragment>
`;
